cmd/opm/generate/cmd.go

```diff
@@ -71,19 +83,24 @@ value of each duplicate key will be added to the generated Dockerfile.
 			defer f.Close()
 
 			gen := action.GenerateDockerfile{
-				BaseImage:   baseImage,
-				IndexDir:    indexName,
-				ExtraLabels: extraLabels,
-				Writer:      f,
+				BaseImage:    baseImage,
+				BuilderImage: builderImage,
+				IndexDir:     indexName,
+				ExtraLabels:  extraLabels,
+				Writer:       f,
 			}
 			if err := gen.Run(); err != nil {
 				log.Fatal(err)
 			}
 			return nil
 		},
 	}
-	cmd.Flags().StringVarP(&baseImage, "binary-image", "i", containertools.DefaultBinarySourceImage, "Image in which to build catalog.")
+	cmd.Flags().StringVar(&baseImage, "binary-image", containertools.DefaultBinarySourceImage, "Image in which to build catalog.")
+	cmd.Flags().StringVarP(&baseImage, "base-image", "i", containertools.DefaultBinarySourceImage, "Image base to use to build catalog.")
+	cmd.Flags().StringVarP(&builderImage, "builder-image", "b", containertools.DefaultBinarySourceImage, "Image to use as a build stage.")
 	cmd.Flags().StringSliceVarP(&extraLabelStrs, "extra-labels", "l", []string{}, "Extra labels to include in the generated Dockerfile. Labels should be of the form 'key=value'.")
+	cmd.Flags().MarkDeprecated("binary-image", "use --base-image instead")
+	cmd.MarkFlagsMutuallyExclusive("binary-image", "base-image")
```

Comment:
Where did we land on adding this?
```
cmd.MarkFlagsMutuallyExclusive("binary-image", "builder-image")
```

Reply:
I thought we said that we'd allow mixing them, while complaining. 
So someone gives us binary-file and we go "that's deprecated", but we also use it appropriately. 
The only special handling is if they use `binary-image` and nothing else, where we use that value for builder AND base.
But I'm not strongly opinionated on this, so let me know if you are.